### PR TITLE
H218: Mirror p=0.5 + tau_y=2.0 — untested (p × tau_y) grid cell

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis H218: Mirror p=0.5 + tau_y=2.0 — untested (p × tau_y) grid cell

**Advisor branch:** tay  
**Student:** nezuko  
**Hypothesis ID:** H218

---

### Background

Finding S (H212/fern) showed that `--mirror-augment-p 0.5` alone collapses `delta_mirror_WSS` (the mirror-invariance score) rather than improving it — the model memorises the augmented view and loses physical antisymmetry. At the same time, Finding H (H168/alphonse) confirmed that `--tau-y-loss-weight 2.0` improves VP/SP without harming WSS.

**The open question:** does increasing tau_y loss weight _while_ applying p=0.5 mirror augmentation rescue the WSS antisymmetry collapse? The (p=0.5 × tau_y=2.0) cell of the grid has never been tested. Theory says the extra pressure gradient supervision could anchor the flow field enough that p=0.5 does not cause collapse.

### Hypothesis

> Combining `--mirror-augment-p 0.5` with `--tau-y-loss-weight 2.0` will avoid the mirror-invariance collapse observed with p=0.5 alone while retaining the VP/SP benefit of tau_y=2.0, yielding a net improvement in `val_abupt`.

### Starting checkpoint

Resume from the current SOTA: **W&B run `bx3t1vdw`** (EP13 EMA, H209).

```
--wandb_run_id bx3t1vdw
```

### Experiment

Train for **3 epochs** (EP14–EP16) from the H209 checkpoint with both flags active simultaneously:

```bash
torchrun --standalone --nproc-per-node=8 target/train.py \
  --wandb_run_id bx3t1vdw \
  --mirror-augment-p 0.5 \
  --tau-y-loss-weight 2.0 \
  --wandb_group h218-p05-tauy2 \
  $SENPAI_EXTRA_ARGS
```

After training, evaluate with TTA:

```bash
python target/eval_tta_h209.py --wandb_run_id <your_run_id>
```

### Current SOTA baseline (target to beat)

| Metric | Value |
|---|---|
| val_abupt | 5.9755 |
| test_abupt | 5.8221 |
| test_VP | 3.421 |
| test_SP | 3.577 |
| test_WSS | 6.727 |

**SOTA gate:** `val_abupt < 5.9755 AND test_abupt < 5.8221`

### What to watch

- `delta_mirror_WSS`: Did the collapse from p=0.5 (Finding S) happen again, or did tau_y=2.0 anchor it?
- `test_WSS` vs baseline 6.727 — collapse would push this up
- `val_abupt` and `test_abupt` vs SOTA gate

### Expected outcome

If delta_mirror_WSS stays healthy (≥ baseline) and val_abupt improves, we have found a safe (p, tau_y) combination and the grid should be expanded. If collapse still occurs, the tau_y anchor is insufficient and p=0.5 should be considered permanently off-limits.

### Instructions to student

1. Resume from checkpoint `bx3t1vdw`.
2. Run training exactly as shown above for 3 epochs.
3. Run TTA eval after training.
4. Report `val_abupt`, `test_abupt`, `test_VP`, `test_SP`, `test_WSS`, and `delta_mirror_WSS` in a PR comment.
5. Post the terminal `SENPAI-RESULT` marker once complete.

```
SENPAI-RESULT: {"terminal":true,"status":"complete","pending_arms":false,"wandb_run_ids":["<run-id>"],"primary_metric":{"name":"val_abupt","value":<number>},"test_metric":{"name":"test_abupt","value":<number>}}
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT nezuko: Infrastructure gap analysis before launch.

**Issues with PR command as stated:**

1. `--wandb_run_id bx3t1vdw` doesn't exist on `tay` — there's no checkpoint-resume/wandb-run-resume mechanism in `train.py` or `trainer_runtime.py`. Confirmed by grep across both files.
2. `--mirror-augment-p 0.5` doesn't exist on `tay` — the H148/H183/H190 mirror-augmentation flags were never merged from those side branches into `tay`.
3. `bx3t1vdw` is the H209 TTA *eval* run; the actual underlying trained model is H185 W&B run `yw2a5dyl` (verified — artifact `model-thorfinn-h185-h171-mirror-aug-compound-yw2a5dyl:v0` with `best`/`epoch-13` aliases is accessible).
4. The H185 source run config shows `mirror_augmentation: True` (H148 boolean style, no prob arg → effectively p=0.5) and `tau_y_loss_weight: 1.3` (NOT `tau_y=3.0` as BASELINE.md states). The BASELINE.md "mirror p=0.25 + tau_y=3.0" description doesn't match the source config.

**Plan I'm proceeding with** (will deviate from PR command but preserve hypothesis intent):
- Port H183/H191 `--mirror-augmentation` flag implementation into `train.py` (boolean; defaults to p=0.5 per H148 semantics).
- Add `--load-checkpoint-path` + `--resume-epoch-start` mechanism to `train.py` to load H185 EP13 EMA weights, initialize EMA shadow from those weights, and continue training for **3 more epochs** with the same H185 recipe except `--tau-y-loss-weight 2.0` (vs H185's 1.3). Mirror augmentation remains on (already on in H185).
- Download `yw2a5dyl` best artifact via wandb.Api (already proven pattern from `ensemble_eval.py:70-85`).
- Run TTA eval via `eval_tta_h209.py` on the resulting EP16 EMA checkpoint.

**Hypothesis-equivalent interpretation:** since H185 *already* trained with mirror p=0.5 (boolean default), the test is effectively whether bumping `tau_y` from 1.3 → 2.0 in 3 fine-tune epochs (a) helps val_abupt/test_abupt cross the SOTA gate and (b) preserves `delta_mirror_WSS`. If the advisor wants a different interpretation, please leave a note and I'll adjust.

**Cross-reference:** H191 (PR #1365) already tested mirror+tau_y=2.0 from scratch with H112 recipe → val_abupt=6.2304 / test_abupt=5.9616 / test_WSS=6.8948, WSS_x slope NEGATIVE (basin intact). The novel contribution of H218 here is the resume-from-converged-checkpoint angle.

Launching implementation now; will report progress.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #1394 because ADVISOR — CLOSING WITH APOLOGY. The Round 4 assignment had two systemic errors that I (advisor) own:

**Error 1: Recipe spec did not match actual yw2a5dyl config.** I cited the H185 recipe as `tau_y=3.0, tau_z=2.0, --mirror-augment-p 0.25, --grad-clip 1.0, --lr-schedule cosine, --enable-residual-positions, --save-best-checkpoint`. Verified from W&B config on yw2a5dyl, the ACTUAL recipe is `tau_y=1.3, tau_z=1.67, mirror_augmentation=True (boolean), --grad-clip-norm 0.5, --lr-cosine-t-max 13`. The other flags don't exist on tay. Student catches were correct.

**Error 2: Budget infeasibility.** yw2a5dyl actual `total_train_minutes=874.4` (~14.6h on 8 GPUs). The 6h SENPAI_TIMEOUT_MINUTES cap CANNOT fit a 13-epoch H185-recipe retrain. The Round 4 assignment was incompatible with the budget cap from the start.

**Mirror augmentation also not on tay:** verified via grep — `--mirror-augmentation` lives on `askeladd/h148-mirror-augmentation` and `fern/h183-mirror-aug-tau-y-3p0-compound` branches, never merged. Reproducing H185 recipe requires cherry-pick PR first.

BASELINE.md has been corrected on tay (commit pending).

**Reassignment incoming**: H228 — 4-pass TTA: orig + mirror + rot_x+2° + rot_x−2° on H185 — explicit stacking with askeladd's rotation (eval-only, ~45min)

Thank you for catching the recipe discrepancies before launching. Apologies for the wasted setup time..
